src/components/SQForm/SQFormMultiValue.tsx

```diff
@@ -250,76 +288,61 @@ function SQFormMultiValue({
             );
           });
         }}
-        ListboxComponent={ListboxVirtualizedComponent}
-        getOptionLabel={option => option.label || ''}
+        ListboxComponent={
+          ListboxVirtualizedComponent as React.ComponentType<
+            React.HTMLAttributes<HTMLElement>
+          >
+        }
+        getOptionLabel={(option) => option.label || ''}
         value={fieldValue || []}
         inputValue={inputValue}
         onBlur={handleAutocompleteBlur}
         onChange={handleAutocompleteChange}
         onInputChange={handleInputChange}
-        getOptionDisabled={option => option.isDisabled}
+        getOptionDisabled={(option) => option.isDisabled}
         disabled={isDisabled}
         disableClearable={isDisabled}
-        renderInput={params => (
-          <TextField
-            {...params}
-            name={name}
-            color="primary"
-            disabled={isDisabled}
-            fullWidth={true}
-            label={label}
-            InputProps={{
-              ...params.InputProps,
-              endAdornment: {
-                ...params.InputProps.endAdornment,
-                props: {
-                  ...params.InputProps.endAdornment.props,
-                  className: `${params.InputProps.endAdornment.props.className} ${clearButtonClasses.clearButton}`
-                }
-              }
-            }}
-            InputLabelProps={{
-              ...params.InputLabelProps,
-              shrink: true
-            }}
-            inputProps={{
-              ...params.inputProps,
-              disabled: isDisabled
-            }}
-            FormHelperTextProps={{error: isFieldError}}
-            helperText={!isDisabled && HelperTextComponent}
-          />
-        )}
+        renderInput={(params) => {
+          console.log('params', params);
+          // const endAdornment = params.InputProps.endAdornment || null;
+          const endAdornment = Object.assign<
+            Record<string, Record<string, unknown>>,
+            Record<string, Record<string, unknown>>
+          >({}, params.InputProps.endAdornment || {});
+          if (endAdornment && endAdornment.props) {
+            endAdornment.props = {
+              ...(endAdornment.props || {}),
+              className: `${endAdornment.props.className} ${clearButtonClasses.endAdornment}`,
+            };
+          }
```

Comment:
I'm not so sure about this, particularly the type for `Object.assign`.

Reply:
Oof, yeah it's kind of hard for me to follow what's going on with that at the moment, but maybe the `InputAdornmentProps` type could help here?

Reply:
So I was able to come up with a different solution that is far easier to read and gets the job done